### DISCUSSION TITLE
explicit systems optional?

### DISCUSSION THREAD

**Discussion by edrex** [hercules-ci/flake-parts#128] [Ideas]
Looking for ways to increase the ratio of meaningful choices to boilerplate in flake.nix definitions.

For me, the value of `systems` is nearly always an arbitrary choice.

Any reason not to provide an inclusive default value, given that evaluation is lazy?

**Comment by roberth** [comment]
Perhaps nixpkgs could set a default.
There's some work in progress here
 - https://github.com/NixOS/nixpkgs/pull/222155

**Reply by edrex** [reply]
Ohhh right, `systems` is a flakeModule option so a default could be provided by any module. I'm still getting the hang of the module system. Pardon my slow uptake.

**Reply by srid** [reply]
@edrex See also https://github.com/hercules-ci/flake-parts/issues/25

**Reply by edrex** [reply]
> A default value for systems might be a good idea

**Reply by edrex** [reply]
Robert, what would be the reasoning for putting the default into the nixpkgs module rather than in a core module, given that `systems` isn't specific to nixpkgs (other than the fact that `lib.systems.flakeExposed` lives there)?

**Reply by edrex** [reply]
My only concern would be that if the nixpkgs module isn't needed for some other reason (adding an overlay) it's an extra line in the minimal flake definition. But maybe you're envisioning the nixpkgs module being somewhat mandatory?

**Reply by roberth** [reply]
My preferred solution is to make the enumeration of systems completely unnecessary in the Nix CLI schema, but that's a significant change that won't be carried out soon, so short of that, I think a default in `nixpkgs.flakeModule` would be a sensible, pragmatic solution.

**Comment by edrex** [comment]
Workaround for ordinary flake authors like me who aren't keen to participate in a meta-architectural deadlock: https://github.com/edrex/lab_journal/commit/66f22782b171f0b825df74103fca61ac54c901a8
